Description: Fixed incorrect princomp(cx_mat,cx_mat) calls
Forwarded: not-needed
Origin: https://gitlab.com/conradsnicta/armadillo-code/-/commit/d9002c70fb69cb3d0653ace4045d812a4a4ab35b

diff -Naur armadillo-10.1.2/include/armadillo_bits/op_princomp_meat.hpp armadillo-10.1.2-princomp_fix/include/armadillo_bits/op_princomp_meat.hpp
--- armadillo-10.1.2/include/armadillo_bits/op_princomp_meat.hpp	2016-06-17 02:22:12.000000000 +1000
+++ armadillo-10.1.2-princomp_fix/include/armadillo_bits/op_princomp_meat.hpp	2016-06-17 02:22:13.000000000 +1000
@@ -264,6 +264,7 @@
   arma_extra_debug_sigprint();
   
   typedef typename T1::elem_type eT;
+  typedef typename T1::pod_type   T;
   
   const unwrap<T1>    Y( X.get_ref() );
   const Mat<eT>& in = Y.M;
@@ -274,7 +275,7 @@
     
     // singular value decomposition
     Mat<eT> U;
-    Col<eT> s;
+    Col< T> s;
     
     const bool svd_ok = (in.n_rows >= in.n_cols) ? svd_econ(U, s, coeff_out, tmp) : svd(U, s, coeff_out, tmp);
     
