Description: Fixed incorrect princomp(cx_mat,cx_mat) calls
Forwarded: not-needed
Author: Conrad Sanderson <conradsand@gmail.com>

diff -Naur armadillo-10.1.2/include/armadillo_bits/op_pinv_meat.hpp armadillo-10.1.2-pinv-bugfix/include/armadillo_bits/op_pinv_meat.hpp
--- armadillo-10.1.2/include/armadillo_bits/op_pinv_meat.hpp	2016-06-17 02:22:12.000000000 +1000
+++ armadillo-10.1.2-pinv-bugfix/include/armadillo_bits/op_pinv_meat.hpp	2021-03-09 16:25:49.909112061 +1000
@@ -119,7 +119,7 @@
       {
       const T val = s_mem[i];
       
-      if(val >= tol)  {  s2_mem[count2] = T(1) / val;  ++count2; }
+      if(val >= tol)  { s2_mem[count2] = (val > T(0)) ? T(T(1) / val) : T(0); ++count2; }
       }
     
     
