diff -r -u armadillo-8.400.0/include/armadillo_bits/band_helper.hpp armadillo-8.400.0-fix/include/armadillo_bits/band_helper.hpp
--- armadillo-8.400.0/include/armadillo_bits/band_helper.hpp	2016-06-17 02:17:40.000000000 +1000
+++ armadillo-8.400.0-fix/include/armadillo_bits/band_helper.hpp	2016-06-17 02:17:42.000000000 +1000
@@ -80,8 +80,8 @@
     
     if( (L_count > KL) || (U_count > KU) )
       {
-      KL = L_count;
-      KU = U_count;
+      KL = (std::max)(KL, L_count);
+      KU = (std::max)(KU, U_count);
       
       const uword n_nonzero = N*(KL+KU+1) - (KL*(KL+1) + KU*(KU+1))/2;
       
